tools/lib/provision_inner.py

```diff
@@ -198,15 +221,12 @@ def main(options: argparse.Namespace) -> int:
             destroy_leaked_test_databases,
         )
 
-        try:
-            from zerver.lib.queue import SimpleQueueClient
-            SimpleQueueClient()
-            rabbitmq_is_configured = True
-        except Exception:
-            rabbitmq_is_configured = False
-
-        if options.is_force or not rabbitmq_is_configured:
+        if options.is_force or need_to_run_configure_rabbitmq():
             run(["scripts/setup/configure-rabbitmq"])
+            write_new_digest(
```

Comment:
@showell nice catch! This was making us rebuild the entire database on the 2nd provision since we didn't write a new digest after configuring rabbitmq.

Reply:
I don't understand why this would cause that -- there was previously no digest at all for rabbitmq; it just checked by attempting to connect to rabbitmq whether the service was available.

Reply:
We have the old digest from the 1st provision attempt.
I checked, without writing a new digest, on the 2nd provision, `DEV_DATABASE.template_status()` returns `needs_rebuild` which makes sense. If we write a new digest, it returns `current`.

Reply:
But the digest I added here only affects rabbitmq.  I am confused about your comments here.

Reply:
oh! I haven't tested on this PR. This means we are still missing writing a digest after the migrations via this:
```
    def write_new_migration_digest(self) -> None:
        write_new_digest(
            self.migration_digest_file,
            migration_paths(),
        )
```
I suppose. You can test, If you run provision twice on a fresh zulip clone, on a fresh OS, it will rebuild db twice.

Reply:
Sorry for confusion here, I initially had thought, digests were used for a singular purpose.